system/jlib/jfile.cpp

```diff
@@ -7947,38 +7933,75 @@ MODULE_INIT(INIT_PRIORITY_STANDARD)
-            values[BlockedSequentialIO] = (unsetPlaneAttrValue != value) ? value * 1024 : value;
-            value = plane.getPropInt64(("@" + std::string(planeAttributeTypeStrings[BlockedRandomIO])).c_str(), unsetPlaneAttrValue);
-            values[BlockedRandomIO] = (unsetPlaneAttrValue != value) ? value * 1024 : value;
-
-            // plane expert settings
-            std::string propName = "expert/@" + std::string(planeAttributeTypeStrings[FileSyncWriteClose]);
-            if (plane.hasProp(propName.c_str()))
-                values[FileSyncWriteClose] = plane.getPropBool(propName.c_str()) ? 1 : 0;
-            else
+            for (unsigned propNum=0; propNum<PlaneAttributeType::PlaneAttributeCount; ++propNum)
             {
-                // temporary (check legacy fileSyncMaxRetrySecs too), purely for short term backward compatibility (see HPCC-xxxx)
-                unsigned __int64 v = plane.getPropInt64("expert/@fileSyncMaxRetrySecs", unsetPlaneAttrValue);
-                // NB: fileSyncMaxRetrySecs==0 is treated as set/enabled
-                values[FileSyncWriteClose] = v != unsetPlaneAttrValue ? 1 : 0;
+                const PlaneAttributeInfo &attrInfo = planeAttributeInfo[propNum];
+                std::string prop;
+                if (attrInfo.isExpert)
+                    prop += "expert/";
+                prop += "@" + std::string(attrInfo.name);
+                switch (attrInfo.type)
+                {
+                    case PlaneAttrType::integer:
+                    {
+                        unsigned __int64 value = plane.getPropInt64(prop.c_str(), unsetPlaneAttrValue);
+                        if (unsetPlaneAttrValue != value)
+                        {
+                            if (attrInfo.scale)
+                            {
+                                dbgassertex(PlaneAttrType::integer == attrInfo.type);
+                                value *= attrInfo.scale;
+                            }
+                        }
+                        values[propNum] = value;
+                        break;
+                    }
+                    case PlaneAttrType::boolean:
+                    {
+                        unsigned __int64 value;
+                        if (plane.hasProp(prop.c_str()))
+                            value = plane.getPropBool(prop.c_str()) ? 1 : 0;
+                        else if (FileSyncWriteClose == propNum) // temporary (if FileSyncWriteClose and unset, check legacy fileSyncMaxRetrySecs), purely for short term backward compatibility (see HPCC-32757)
+                        {
+                            unsigned __int64 v = plane.getPropInt64("expert/@fileSyncMaxRetrySecs", unsetPlaneAttrValue);
+                            // NB: fileSyncMaxRetrySecs==0 is treated as set/enabled
+                            if (unsetPlaneAttrValue != v)
```

Comment:
@ghalliday - as you pointed out before I think, this was wrong before this final commit. As it was, it caused the value to be 0 if not set (which is the default anyway, but..)